app/src/main/java/com/vero/tasky/agenda/presentation/agendaevent/EventDetailsViewModel.kt

```diff
@@ -0,0 +1,226 @@
+) : ViewModel() {
+
+    private val userId = userPreferences.getUser()?.userId ?: ""
+    private val itemId = savedStateHandle.get<String?>(NavigationConstants.ITEM_ID)
+
+    var state by mutableStateOf(
+        (savedStateHandle.get(STATE_KEY) as? EventDetailsState)?.copy(
+            isLoading = false
+        ) ?: EventDetailsState(
+            isEditable = savedStateHandle[NavigationConstants.IS_EDITABLE] ?: false,
+            isNewEvent = itemId != null
+        )
+    )
+        private set
+
+    private val channel = Channel<UiEventDetailsEvent>()
+    val uiEvent = channel.receiveAsFlow()
+
+
+    init {
+        itemId?.let { id ->
+            eventUseCases.getEvent.invoke(id).onEach { agendaItem ->
+                updateState(
+                    state.copy(
+                        agendaItem = agendaItem,
+                        isGoing = if (agendaItem.isUserEventCreator) true
+                        else agendaItem.attendees.find { userId == it.userId }?.isGoing ?: false,
+                    )
+                )
+            }.launchIn(viewModelScope)
+        }
+    }
+
+    fun onEvent(event: EventDetailsEvent) {
+        when (event) {
+            is EventDetailsEvent.AddAttendee -> addAttendee(event.email)
+            is EventDetailsEvent.AddPhoto -> addPhoto(event.uri)
+            EventDetailsEvent.ChangeIsGoingState -> changeIsGoingState()
+            EventDetailsEvent.ChangeMode -> changeMode()
+            EventDetailsEvent.CreateEvent -> createEvent()
+            EventDetailsEvent.DeleteEvent -> deleteEvent()
+            is EventDetailsEvent.FromDateTimeChanged -> changeFromDateTime(event.time)
+            is EventDetailsEvent.ReminderChanged -> changeReminder(event.reminderRange)
+            is EventDetailsEvent.ToDateTimeChanged -> changeToDateTime(event.time)
+            EventDetailsEvent.UpdateEvent -> updateAgendaItemEvent()
+            is EventDetailsEvent.DeletePhoto -> deletePhoto(event.key)
+        }
+    }
+
+    private fun deletePhoto(key: String) {
+        val deletedPhotoKeys = mutableListOf<String>()
+        deletedPhotoKeys.addAll(state.deletedPhotoKeys)
+        deletedPhotoKeys.add(key)
+        updateState(state.copy(deletedPhotoKeys = deletedPhotoKeys))
+    }
+
+    private fun updateAgendaItemEvent() {
+        viewModelScope.launch {
+            eventUseCases.updateEvent(
+                event = state.agendaItem,
+                deletedPhotoKeys = state.deletedPhotoKeys,
+                isGoing = state.isGoing
+            ).onSuccess {
+                channel.send(UiEventDetailsEvent.ShowInfoToast(it.countOfSkippedMessages.toString()))
+                channel.send(UiEventDetailsEvent.OnBackClick)
+            }.onFailure {
+                channel.send(UiEventDetailsEvent.ShowErrorMessage(R.string.default_event_details_error))
+            }
+        }
+    }
+
+    private fun changeIsGoingState() {
+        viewModelScope.launch {
```

Comment:
You can call the UpdateEvent method you created on line 88
Instead of duplicating the code